containers/Dockerfile.buildenv1804

```diff
@@ -0,0 +1,229 @@
+# Distributed under the MIT License.
```

Comment:
I worry this buildenv is going to become out-of-date and dysfunctional quite soon if we don't test it on CI. WDYT?

Reply:
Could you please use words? I don't understand the question

Reply:
Sorry, that acronym is supposed to mean "What do you think?" :)

Reply:
Ah! Okay :) It's not entirely clear to me. For the most part the build env has been very stable, but I agree that it almost certainly wouldn't see updates. My reason for not testing in CI was that I didn't know if there was a way of having a second docker image without duplicating a lot of our Linux build instructions. However, it might be worth making that duplication :) So in short, I have no strong feelings either way :)